code/processes/dqe.q

```diff
@@ -44,29 +44,35 @@ fillprocname:{[rs;h]
   (flip a),val,'`
   }
 
-initstatusupd:{[id;funct;vars;rs]                                                                               /- set initial values in results table
-  .lg.o[`initstatus;"setting up initial record(s) for id ",(string id)];
-  `.dqe.results insert (id;funct;`$"," sv string raze (),vars;rs[0];rs[1];.z.p;0Np;0b;"";`started);
+dupchk:{[idnum;proc]                                                                                            /- checks for unfinished runs that match the new run
+  if[count select from .dqe.results where id=idnum,procschk=proc,chkstatus=`started;
+    .dqe.failchk[idnum;"error:fail to complete before next run";proc]];
+  }
+
+initstatusupd:{[idnum;funct;vars;rs]                                                                            /- set initial values in results table
+  .lg.o[`initstatus;"setting up initial record(s) for id ",(string idnum)];
+  .dqe.dupchk[idnum]'[rs];
+  `.dqe.results insert (idnum;funct;`$"," sv string raze (),vars;rs[0];rs[1];.z.p;0Np;0b;"";`started);
   }
 
 failchk:{[idnum;error;proc]
-  c:count select from .dqe.results where id=idnum, procschk=proc;
-  .lg.o[`failerr;raze "run check id ",(string idnum)," update in results table with a fail, with ",(string error)];
-  `.dqe.results set update chkstatus:`failed,output:0b,descp:c#enlist error from .dqe.results where id=idnum, procschk=proc;
+  c:count select from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;
+  if[c;.lg.o[`failerr;raze "run check id ",(string idnum)," update in results table with a fail, with ",(string error)]];
+  `.dqe.results set update chkstatus:`failed,output:0b,descp:c#enlist error from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;
   }
 
 postback:{[idnum;proc;result]
   $["e"=first result;
   .dqe.failchk[idnum;result;proc];
-  `.dqe.results set update endtime:.z.p,output:`boolean$ string first result,descp:enlist last result,chkstatus:`complete from .dqe.results where id=idnum,procschk=proc];
+  `.dqe.results set update endtime:.z.p,output:`boolean$ string first result,descp:enlist last result,chkstatus:`complete from .dqe.results where id=idnum,procschk=proc,chkstatus=`started];
```

Comment:
What does `result` look like? Can we remove the casting here by solving this problem at an earlier stage?